VogtPayroll7/Payroll.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VogtPayroll7
+{
+    class Payroll
+    {
+        private List<Employee> _employees = new List<Employee>();
+
+        public void AddEmployees(List<Employee> empList)
+        {
+            _employees.AddRange(empList);
+        }
+
+        public int GetEmployeeCount()
+        {
+            Console.WriteLine($"Employee count is: {_employees.Count}");
+            return _employees.Count;
+        }
+
+        public void GetBonusAmount()
+        {
+            foreach (var employee in _employees)
+            {
+                if (employee.YearsOfService >= 1 && employee.YearsOfService <= 4)
```

Comment:
This logic should live within the Employee fn. Encapsualte that logic within the class itself.

However, you shouldn't be storing this in a class variable. Instead, create a function so that you can calculate the bonus on demand/whenever you need it. Then, get rid of this function here in Payroll.cs entirely.

So for example, your **employee** function should look more like this:
```c#
public int GetBonusAmount() {
        if (employee.YearsOfService >= 1 && employee.YearsOfService <= 4)
                {
                    return 100m;
                }
                else if (employee.YearsOfService >= 4 && employee.YearsOfService <= 9)
                {
                    return 500m;
                }
                else if(employee.YearsOfService >= 10)
                {
                    return 1000m;
                }
                else
                {
                    Console.WriteLine($"ERROR: Failed to calculate bonus for {_empName}. Years need to be greater than equal to one. Actual: {BonusAmount}");
                }
```

Then, you can use that function here:
```c#
        public decimal GetTotalAmountAwardedForAllEmployees()
        {
            decimal totalAmount = 0m;

            GetBonusAmount();

            foreach (var employee in _employees)
            {
                totalAmount += employee.GetBonusAmount();
            }
```